### DISCUSSION TITLE
Universe 2023: The best developer experience built by developers, for developers - DEV2727M

### DISCUSSION THREAD

**Discussion by jricciardi** [community/community#74930] [Other Feature Feedback, Questions, & Ideas]
Thursday, Nov 910:15 AM - 10:55 AM PST
From the first line of code that was ever committed to build GitHub, our purpose has been to equip developers with the tools they need to do their best work on a platform purpose-built for them. In this talk Mario Rodriguez, VP of Product Management at GitHub, will be diving in and sharing all of the exciting new features across our platform that increase developer productivity, boost creative collaboration, and ensure that organizations have the features and functionality they need to innovate at scale. GitHub—the best developer experience built by developers, for developers.

Mario Rodriguez
VP Product Management, GitHub

> Mario Rodriguez is a VP of Product Management at GitHub, currently focusing on all things Productivity. He oversees Repos, Pull Requests, Issues, Projects, Mobile and our AI strategy including Copilot. Mario's core identity is being a learner and time away from product engineering is spent with his wife and two daughters.

**Comment by jricciardi** [comment]
Welcome to the stage, Mario! 🚀 
![Screenshot 2023-11-09 at 10 16 05 AM](https://github.com/community/community/assets/31521880/944cee66-f439-4037-8018-d15cdd7b92b4)

![Screenshot 2023-11-09 at 10 17 14 AM](https://github.com/community/community/assets/31521880/72ab095e-904c-46d8-a8f7-697a0f3eaaf5)

**Comment by jricciardi** [comment]
Was "Shift Left" on the [Universe Bingo card?](https://github.com/orgs/community/discussions/74691) Security is a HUGE theme this year (share the road, Copilot 😉 )

![Screenshot 2023-11-09 at 10 18 30 AM](https://github.com/community/community/assets/31521880/9b4a19a7-1338-4584-9563-13fa3fad0348)

**Comment by jricciardi** [comment]
Asking Copilot for jokes! That's a live demo for ya
![Screenshot 2023-11-09 at 10 25 52 AM](https://github.com/community/community/assets/31521880/c45f0e48-19ae-4c44-9353-79965d0cf702)

**Reply by jricciardi** [reply]
Copilot is way better at code and interpreting your repo context than it is at standup comedy. Super cool demo of asking Copilot about Issues and deployments in a repo 👀 

![Screenshot 2023-11-09 at 10 28 14 AM](https://github.com/community/community/assets/31521880/8503b576-fdb4-4ee1-936f-33d5c03ffc52)

**Comment by jricciardi** [comment]
I'm personally so excited about this part of what's next for Copilot: Copilot chat that has specific documentation knowledge, with in-line citations 😍 

![Screenshot 2023-11-09 at 10 30 20 AM](https://github.com/community/community/assets/31521880/72c229d5-b186-4269-aa10-bd2f803e9126)

**Reply by gaurav-ketkar** [reply]
I LOVE that there are inline captions. It helps if I want to dig more/verify what it is telling me is correct and usually is helpful when sharing these answers with others - sending the actual doc link will be considered more authoritative than just the conversational answer from Copilot.

**Reply by jricciardi** [reply]
Totally. Those citations do a lot of work to make suggestions feel more trustworthy and easier to interrogate. Without them, sometimes I just end up asking a model and then going to Google anyways 😅

**Comment by jricciardi** [comment]
Repo Rulesets are also once of those low-key super cool quality of life upgrades for Repositories, especially for maintainers 🙌  More details here: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets

![Screenshot 2023-11-09 at 10 35 17 AM](https://github.com/community/community/assets/31521880/dbd1ea7c-145d-483f-af97-e7b1361cc4bc)
![Screenshot 2023-11-09 at 10 36 00 AM](https://github.com/community/community/assets/31521880/88a975cd-d597-4485-9414-121749159b16)

**Comment by jricciardi** [comment]
"DevEx is not features... it's Flow" - Mario Rodriguez

![Screenshot 2023-11-09 at 10 40 01 AM](https://github.com/community/community/assets/31521880/5124d37f-5532-489a-a7f5-6d0a732d7bff)

**Comment by jricciardi** [comment]
Welcome to Naresh Choudhary, Vice President, Quality & Productivity at Infosys Ltd. for a sit down chat with Mario on stage!

![Screenshot 2023-11-09 at 10 41 27 AM](https://github.com/community/community/assets/31521880/35271526-0fd5-43ce-8d20-25e5072056ce)
![Screenshot 2023-11-09 at 10 42 59 AM](https://github.com/community/community/assets/31521880/31be60b0-f497-40c5-b613-1627da4be749)



Naresh Choudhary
Vice President, Quality & Productivity, Infosys Ltd.

> Naresh Choudhary leads the effort on Engineering Excellence, Building and Deploying Enterprise Platforms for Developer Productivity & Quality across Agile & DevSecOps, Open Source, Cloud, AI, Quality Assurance, Driving collaboration with Infosys’ key technology partners and leading several Innovation, Change programs as part of the Infosys Live Enterprise.

**Comment by jricciardi** [comment]
"The use of AI will just become part of our day to day lives ... until it won't feel like 'AI' any more" - Naresh Choudhary

![Screenshot 2023-11-09 at 10 55 24 AM](https://github.com/community/community/assets/31521880/5bc87e66-64b1-46ea-a5a4-f04f72c425a0)

**Comment by jricciardi** [comment]
That's a wrap on the live session! We'll post a link to the keynote recording as soon as the recording is available, sometime later today 🎉

**Comment by SanjayKhatiChhetri** [comment]
This was very insightful uses cases of copilot

**Comment by starspace1** [comment]
I'm impressed w/ Mario's ability to go straight from a live demo to an in depth interview!

**Comment by starspace1** [comment]
I'd love to learn more about the DSO maturity model and scores/metrics Naresh mentioned!
